payjoin-cli/src/app/v2.rs

```diff
@@ -306,57 +305,18 @@ impl App {
     }
 }
 
-async fn unwrap_ohttp_keys_or_else_fetch(config: &AppConfig) -> Result<payjoin::OhttpKeys> {
+async fn unwrap_ohttp_keys_or_else_fetch(config: AppConfig) -> Result<payjoin::OhttpKeys> {
     if let Some(keys) = config.ohttp_keys.clone() {
         Ok(keys)
     } else {
-        fetch_ohttp_keys(&config.ohttp_relay, &config.pj_endpoint).await
+        Ok(tokio::task::spawn_blocking(move || {
+            payjoin_io::fetch_ohttp_keys(config.ohttp_relay.clone(), config.pj_endpoint.clone())
+        })
+        .await
+        .map_err(|e| anyhow!("Failed to fetch ohttp keys: {}", e))??)
     }
 }
```

Comment:
I think we need some `danger-local-https` logic here. Otherwise, if that feature is enabled the `fetch_ohttp_keys` function call has too few arguments and will not compile.

I think you could also get the `Result::Ok` you want out with a couple fewer conversions

```async fn unwrap_ohttp_keys_or_else_fetch(config: AppConfig) -> Result<payjoin::OhttpKeys> {
    if let Some(keys) = config.ohttp_keys.clone() {
        Ok(keys)
    } else {
        tokio::task::spawn_blocking(move || {
            payjoin_io::fetch_ohttp_keys(config.ohttp_relay.clone(), config.pj_endpoint.clone())
        })
        .await?
        .map_err(|e| anyhow!("Failed to fetch ohttp keys: {}", e))
    }
}